src/target/riscv_debug.c

```diff
@@ -942,6 +961,13 @@ static size_t riscv_build_target_description(
 
 	/* XXX: TODO - implement generation of the FPU feature and registers */
 
+	/* Add main CSR registers*/
+	offset += snprintf(buffer + offset, print_size, "</feature><feature name=\"org.gnu.gdb.riscv.csr\">");
+	for (size_t i = 0; i < ARRAY_LENGTH(riscv_csrs); i++) {
+		offset += snprintf(buffer + offset, print_size,
+			" <reg name=\"%s\" bitsize=\"%u\" type=\"int\" regnum=\"%" PRIu32 "\" group=\"csr\" save-restore=\"no\"/>",
```

Comment:
This string adds quite a lot to the size here - and `type="int"` can be inferred by GDB so can be dropped, as can the `group="csr"` part. `save-restore="no"` should be able to be formatted in by %s by grabbing `gdb_reg_save_restore_strings[GDB_SAVE_RESTORE_NO]` which will serve to further compress this string in the binary.

These are all only suggestions to help save some space as this is technically a correct string as-is, just large.